### PR TITLE
ARMOCP-567: AWS and Azure test configs for clusters heterogeneous in arch and kernel pagesize

### PR DESCRIPTION
**Pull request by aleskandro**
This commit adds an AWS connected and Azure disconnected test configs to run openshift-tests-private against clusters heterogeneous in CPU Architecture and kernel pagesize.

The env consists of:
- 3 arm64 masters
- 2 arm64 workers with 4k pagesize kernel
- 1 arm64 worker with 64k pagesize kernel
- 1 amd64 worker (only 4k pagesize kernel is supported)

Frequency is set to f28 for both the test configs. Upgrade jobs will be added once the config file from 4.15 stable to 4.15 nightly is available.

/cc @lwan-wanglin

### CODE REVIEW

**Comment by aleskandro** [comment]
Also refers MCO-800

**Comment by aleskandro** [comment]
/hold

cc @sergiordlr for visibility.

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-azure-ipi-arm-mixarch-kerneltype-f28-day2-64k-pages

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__multi-nightly.yaml [hunk 1]
```diff
@@ -292,6 +292,28 @@ tests:
     test:
     - ref: cucushift-installer-check-cluster-health
     workflow: cucushift-installer-rehearse-aws-ipi-sdn-mtu-migrate
+- as: aws-ipi-amd-mixarch-f28-day2-64k-pagesize
```
Comment (lwan-wanglin) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__multi-nightly.yaml:295]: should name be `aws-ipi-arm-mixarch-f28-day2-64k-pagesize`?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__multi-nightly.yaml [hunk 2]
```diff
@@ -292,6 +292,28 @@ tests:
...
+  cron: 14 4 13 * *
+  steps:
+    cluster_profile: aws-qe
+    dependencies:
+      OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: release:multi-latest
+    env:
+      ADDITIONAL_WORKER_ARCHITECTURE: x86_64
+      ADDITIONAL_WORKER_VM_TYPE: m6i.xlarge
+      ADDITIONAL_WORKERS: "1"
+      BASE_DOMAIN: qe.devcluster.openshift.com
```
Comment (lwan-wanglin) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__multi-nightly.yaml:305]: missing `COMPUTE_NODE_TYPE: m6g.xlarge` and `OCP_ARCH: arm64` in env section?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__multi-nightly.yaml [hunk 3]
```diff
@@ -601,6 +623,28 @@ tests:
...
+  cron: 14 4 23 * *
+  steps:
+    cluster_profile: azure-qe
+    dependencies:
+      OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: release:multi-latest
+    env:
+      ADDITIONAL_WORKER_ARCHITECTURE: x86_64
+      ADDITIONAL_WORKER_VM_TYPE: Standard_D4s_v3
+      ADDITIONAL_WORKERS: "1"
+      BASE_DOMAIN: qe.devcluster.openshift.com
```
Comment (lwan-wanglin) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__multi-nightly.yaml:636]: missing `COMPUTE_NODE_TYPE: Standard_D4ps_v5` and `OCP_ARCH: arm64`?

**Comment by aleskandro** [comment]
hi @lwan-wanglin thanks for the comments, going to address them and continue here. In the meantime, the azure fullypriv-disc seems permafailing at the mirror-by-oc-adm step. Do you have any info about it? Perhaps, the bastion get full in space?

**Comment by lwan-wanglin** [comment]
> hi @lwan-wanglin thanks for the comments, going to address them and continue here. In the meantime, the azure fullypriv-disc seems permafailing at the mirror-by-oc-adm step. Do you have any info about it? Perhaps, the bastion get full in storage?

I checked azure disconnected jobs in reportportal, no such failures, it might be a flaky, we can rebuild the job.

**Comment by aleskandro** [comment]
- https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/45768/rehearse-45768-periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-azure-ipi-arm-mixarch-disc-fullypriv-f28-day2-64k-pagesize/1727618840881270784
- https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/45768/rehearse-45768-periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-aws-ipi-arm-mixarch-f28-day2-64k-pagesize/1727618840960962560

These runs look successful, I'm rebasing and squashing so that we're ready to merge.

**Comment by aleskandro** [comment]
/unhold

Hi @sergiordlr, any objections to the added tests configs? We want to be as minimal as possible for now, so the idea is to add an AWS-connected scenario and an Azure full disconnected one. BM tests will come later as they are partially blocked by the new infra configuration we are performing.

**Comment by sergiordlr** [comment]
Hello! It looks good to me.

The only thing I could say is that day2 kerneltype config is not exciting the kerneltype bootstrap code, so adding a workflow to boot the cluster directly with day1 64k kernel would increase the coverage. But since we are trying to keep things minimal, imho it is acceptable to skip this scenario, especially when this is already tested via realtime kernel.

**Comment by aleskandro** [comment]
Yes, I did only day2 because the support for multiarch for day1 conf has not been implemented and we would avoid additional single arch configs especially for this scenario that is inherently a "heterogeneous" cluster (in kernel pagesize first). Moreover, covering the bootstrap node seems superfluous for this feature as users shouldn't need 64k pagesize kernels on bootstrap and control plane.

Thanks @sergiordlr 

/assign @liangxia

**Pull request merged**